tests/inverted-index-testSpec.js

```diff
@@ -1,7 +1,201 @@
+           super: [1],
+           man: [1]
+        });
+      });
+
+    it('Should return valid output for multiple files', () => {
+      expect(multipleFiles.createIndex()).toEqual(
+        { 'book.json': 
+          { an: [ 0 ],
+            inquiry: [ 0 ],
+            into: [ 0 ],
+            the: [ 0, 1 ],
+            wealth: [ 0 ],
+            of: [ 0 ],
+            nations: [ 0 ],
+            this: [ 0, 1 ],
+            string: [ 0, 1 ],
+            seeks: [ 0 ],
+            to: [ 0, 1 ],
+            help: [ 0, 1 ],
+            you: [ 0, 1 ],
+            understand: [ 0, 1 ],
+            problem: [ 0, 1 ],
+            set: [ 0, 1 ],
+            from: [ 1 ],
+            third: [ 1 ],
+            world: [ 1 ],
+            first: [ 1 ],
+            is: [ 1 ],
+            also: [ 1 ] },
+          'bookone.json': 
+            { what: [ 0 ],
+              who: [ 0 ],
+              gave: [ 0 ],
+              you: [ 0 ],
+              that: [ 0 ],
+              planet: [ 1 ],
+              crypton: [ 1 ],
+              super: [ 1 ],
+              man: [ 1 ] } 
+            });
+          });
+        });
+
+  describe('Search index', () => {
+    it("Should return 'object' for typeof jsonOfFile.createIndex(fileName, fileContent)", () => {
+      expect(typeof jsonOfFile.searchIndex('An')).toEqual('object');
+    });
+
+    it("Should return { 'third world': [1] } for searchIndex('Third World')", () => {
+      expect(jsonOfFile.searchIndex('third world')).toEqual({ 'third world': [1] });
+    });
+
+    it("Should return { 'third world': [1] } for searchIndex(['Third', 'World'])", () => {
+      expect(jsonOfFile.searchIndex(['third', 'world'])).toEqual({ 'third world': [1] });
+    });
+
+    it("Should return { 'third world': [1] } for searchIndex('Third', 'World')", () => {
+      expect(jsonOfFile.searchIndex('third', 'world')).toEqual({ 'third world': [1] });
+    });
+
+    it('Should return "Search term(s) is not in document" for searchIndex("zod")', () => {
+      expect(jsonOfFile.searchIndex('zod')).toEqual('Search term(s) is not in document');
+    });
+
+    it('Should return { "world": [1] } for searchIndex("zod world")', () => {
+      expect(jsonOfFile.searchIndex('zod world')).toEqual({ 'world': [1] });
+    });
+
+    it('Should return "Search terms(s) is not in document" for searchIndex("planet crypton")', () => {
+      expect(jsonOfFile.searchIndex('planet crypton')).toEqual('Search term(s) is not in document');
+    });
+
+    it('Should return { "world": [1] } for searchIndex("world..")', () => {
+      expect(jsonOfFile.searchIndex('world..')).toEqual({ 'world': [1] });
+    });
+
+    it('Should return {"world": [1]} for searchIndex("WoRld")', () => {
+      expect(jsonOfFile.searchIndex('WoRld')).toEqual({ 'world': [1] });
+    })
+
+    it('Should return {"what": [1]} for searchIndex("what")', () => {
+      expect(validJson.searchIndex('what')).toEqual({ 'what': [ 0 ] });
+    });
```

Comment:
There should be a test case to ensure searchIndex goes through all indexed files if a filename/key is not passed.